internal/grpc/grpc.go

```diff
@@ -34,11 +60,75 @@ var (
 	}`
 
 	defaultCredentials = insecure.NewCredentials()
+
+	_ GrpcConnectionInterface = (*GrpcConnection)(nil)
 )
 
-type wrappedStream struct {
-	grpc.ClientStream
-	*protovalidate.Validator
+func NewGrpcConnection(ctx context.Context, agentConfig *config.Config) (*GrpcConnection, error) {
+	if agentConfig == nil || agentConfig.Command.Server.Type != "grpc" {
+		return nil, fmt.Errorf("invalid command server settings")
```

Comment:
https://github.com/uber-go/guide/blob/master/style.md#errors
```suggestion
		return nil, errors.New("invalid command server settings")
```